recipes/root/all/conanfile.py

```diff
@@ -0,0 +1,300 @@
+import os
+import shutil
+import stat
+from glob import glob
+
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class PythonOption:
+    OFF = "off"
+    SYSTEM = "system"
+    # in future we may allow the user to specify a version when
+    # libPython is available in Conan Center Index.
+    ALL = [OFF, SYSTEM]
+
+
+
+required_conan_version = ">=1.29.1"
+
+class RootConan(ConanFile):
+    name = "root"
+    version = "v6-22-02"
+    license = "LGPL-2.1-or-later"  # of ROOT itself, the recipe is under MIT license.
+    homepage = "https://root.cern/"
+    # ROOT itself is located at: https://github.com/root-project/root
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "CERN ROOT data analysis framework."
+    topics = ("data-analysis", "physics")
+    settings = ("os", "compiler", "build_type", "arch")
+    options = {
+        # Don't allow static build as it is not supported
+        # see: https://sft.its.cern.ch/jira/browse/ROOT-6446
+        # TODO: shared option should be reinstated when hooks issue is resolved
+        # (see: https://github.com/conan-io/hooks/issues/252)
+        # "shared": [True],
+        "fPIC": [True, False],
+        "python": PythonOption.ALL,
+    }
+    default_options = {
+        # "shared": True,
+        "fPIC": True,
+        # default python=off as there is currently no libpython in Conan center
+        "python": PythonOption.OFF,
+    }
+    generators = "cmake_find_package"
```

Comment:
You need to add the `cmake` generator, and use the generated `conanbuildinfo.cmake` inside the cmake script.
This cmake script sets the current C runtime on MSVC (MT vs MD vd MTd vs MDd).
It also sets libstdc++11 vs libstdc++.

You need to do the following in a cmake script:
```cmake
include(conanbuildinfo.cmake)
conan_basic_setup()
```

```suggestion
    generators = "cmake", "cmake_find_package"
```

Reply:
Thanks for your comments. I tried to implement this in 9181355 and c819fc8 but this now fails the CI. I will look into this problem. But it might take me a while to get to this due to the holidays.

Reply:
Happy holidays, and don't hesitate to call for help if you're stuck somewhere.

Reply:
I have looked into this issue.

In 9181355, I patched `conan_basic_setup()` into the main ROOT `CMakeLists.txt`. 
It seems that some parts of the `conan_basic_setup()` conflict with the ROOT CMake causing the build to fail.

Following the documentation at <https://docs.conan.io/en/latest/reference/generators/cmake.html>, in  
8d2a8a2 I instead call: `conan_set_vs_runtime()` directly. 

Does this address your original concern?

Reply:
@uilianries I think your expertise is required 🙊 if you have a chance to checkout this thread that be great!

Reply:
I have investigated what steps inside `conan_basic_setup` were causing build failures. There were two steps causing problems:

1. `conan_output_dirs_setup()`. I disabled this by passing the `NO_OUTPUT_DIRS` argument.

2. `conan_set_libcxx()`. This conflicts with how ROOT's build system automatically sets the relevant flags and (on my Linux gcc system) led to situations where some components were built with the old ABI and some with the new ABI.

There are two possible solutions that I can see:

1. Disable the `conan_set_libcxx()` step by putting `del self.settings.compiler.libcxx` in the recipe and let ROOT handle setting these flags. See: 5dc54946761. I see that this is a very common pattern in recipes in the CCI (appearing in >200 recipes).

2. Raise `ConanInvalidConfiguration` in cases where the Conan setting is different from what ever ROOT chooses. See: f797f50. @SpaceIm points out this commit won't work with Visual Studio so this needs some additional logic to handle other compilers.

Do the reviewers have a preferred solution?

Reply:
Marking this conversation as resolved to keep discussion all in one place. 

This issue is being discussed in the comment thread below at <https://github.com/conan-io/conan-center-index/pull/3732#discussion_r567958665>.